### PR TITLE
Making Plugin compatible with HPOS

### PR DESCRIPTION
**Pull request by dpanta94**
### All Submissions:

* [x] Does your code follow the [Extendables](https://extendomattic.wordpress.com/standardizations/) standards?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changes proposed in this Pull Request:
Making Amazon Pay plugin compatible with HPOS

Made the plugin compatible using this [guide](https://github.com/woocommerce/woocommerce/wiki/High-Performance-Order-Storage-Upgrade-Recipe-Book#declaring-extension-incompatibility)

Skipped:
- includes/compats/class-wc-amazon-payments-advanced-multi-currency-abstract.php L215-216

Closes woo#226 .

### How to test the changes in this Pull Request:

1. [Enable](https://github.com/woocommerce/woocommerce/wiki/High-Performance-Order-Storage-Upgrade-Recipe-Book#how-to-enable-hpos) HPOS in your installation.
2. Plugins functionality should remain the same regardless of HPOS unless you are using legacy API Keys

### Other information:

* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changelog entry

> ADD - Compatibility with HPOS

### CODE REVIEW

File: includes/admin/legacy/class-wc-amazon-payments-advanced-order-admin-legacy.php [hunk 1]
```diff
@@ -108,9 +110,11 @@ public function do_order_action( $order, $id, $action, $version ) {
 	 * @param int $order_id Order ID.
 	 */
 	private function clear_stored_states( $order_id ) {
-		delete_post_meta( $order_id, 'amazon_reference_state' );
-		delete_post_meta( $order_id, 'amazon_capture_state' );
-		delete_post_meta( $order_id, 'amazon_authorization_state' );
+		$order = wc_get_order( $order_id );
```
Comment (gen4ik-saucal) [inline includes/admin/legacy/class-wc-amazon-payments-advanced-order-admin-legacy.php:113]: Maybe check if is WP_ORDER?

**Review by gen4ik-saucal — APPROVED** [review]
Other than that looks good. Good to go.

File: includes/admin/legacy/class-wc-amazon-payments-advanced-order-admin-legacy.php (hunk 1 above)
Comment (dpanta94) [reply includes/admin/legacy/class-wc-amazon-payments-advanced-order-admin-legacy.php:113]: Added!

**Comment by dpanta94** [comment]
@gen4ik-saucal added a few more commits since your last review! COuld you please have another look when possible ?

File: assets/js/amazon-wc-admin.js [hunk 2]
```diff
@@ -366,7 +366,7 @@
 			window.onbeforeunload = null;
 		} );
 	}
-	if ( $( 'body' ).hasClass( 'post-type-shop_order' ) ) {
+	if ( $( 'body' ).hasClass( 'post-type-shop_order' ) || $( 'body' ).hasClass( 'woocommerce_page_wc-orders' ) ) {
```
Comment (gen4ik-saucal) [inline assets/js/amazon-wc-admin.js:369]: Maybe replace it with $( 'body' ).is('.post-type-shop_order','.'woocommerce_page_wc-orders') ?

File: includes/admin/class-wc-amazon-payments-advanced-order-admin.php [hunk 3]
```diff
@@ -60,9 +60,18 @@ public function order_actions_non_ajax() {
 
 		check_admin_referer( 'amazon_order_action', 'security' );
 
-		$order_id = absint( $_GET['post'] ); // TODO: This may break when custom data stores are implemented in the future.
+		$order_id = ! empty( $_GET['post'] ) ? $_GET['post'] : false;
+		if ( ! $order_id ) {
+			$order_id = ! empty( $_GET['id'] ) ? $_GET['id'] : false;
+		}
+
+		$order_id = absint( $order_id );
```
Comment (gen4ik-saucal) [inline includes/admin/class-wc-amazon-payments-advanced-order-admin.php:68]: What if $order_id is still empty?

**Review by gen4ik-saucal — APPROVED** [review]
Just a few minor questions.
It can be merged without changes.

File: includes/admin/class-wc-amazon-payments-advanced-order-admin.php (hunk 3 above)
Comment (dpanta94) [reply includes/admin/class-wc-amazon-payments-advanced-order-admin.php:68]: You are right, amended!

File: assets/js/amazon-wc-admin.js (hunk 2 above)
Comment (dpanta94) [reply assets/js/amazon-wc-admin.js:369]: That didnt work for me :(

Comment (gen4ik-saucal) [reply assets/js/amazon-wc-admin.js:369]: Strange. Ok, leave it as it is.

Good to go.

**Pull request merged**